### PR TITLE
MSEDO-231_Delete_symptom

### PR DESCRIPTION
**Pull request by JoaoLetra**

### CODE REVIEW

File: src/test/javascript/spec/app/entities/symptom/symptom-detail.component.spec.ts [hunk 1]
```diff
@@ -1,17 +1,23 @@
...
-import { Symptom } from 'app/shared/model/symptom.model';
+import {ISymptom, Symptom} from 'app/shared/model/symptom.model';
+import {SymptomService} from "app/entities/symptom/symptom.service";
 
 describe('Component Tests', () => {
   describe('Symptom Management Detail Component', () => {
     let comp: SymptomDetailComponent;
     let fixture: ComponentFixture<SymptomDetailComponent>;
+    let service: SymptomService;
+    let symptom: ISymptom;
```
Comment (andrenns) [inline src/test/javascript/spec/app/entities/symptom/symptom-detail.component.spec.ts:15]: if this is not used please remove (based on github actions warning)

File: src/test/javascript/spec/app/entities/symptom/symptom-detail.component.spec.ts [hunk 2]
```diff
@@ -24,6 +30,7 @@ describe('Component Tests', () => {
         .compileComponents();
       fixture = TestBed.createComponent(SymptomDetailComponent);
       comp = fixture.componentInstance;
+      service = TestBed.inject(SymptomService);
```
Comment (andrenns) [inline src/test/javascript/spec/app/entities/symptom/symptom-detail.component.spec.ts:33]: same for the service here

**Comment by andrenns** [comment]
check warning and correct them if doable

File: src/main/webapp/app/entities/symptom/symptom-detail.component.html [hunk 3]
```diff
@@ -1,10 +1,18 @@
 <div class="row justify-content-center">
     <div class="col-8">
         <div *ngIf="symptom">
+            <button *customHasAnyAuthority="'ROLE_USER'"
+                    type="submit" (click)="delete(symptom)"
+                    class="btn float-right button-transparent "
```
Comment (sfdsilva) [inline src/main/webapp/app/entities/symptom/symptom-detail.component.html:6]: Remove unecessary ` ` on the button class

File: src/main/webapp/app/entities/symptom/symptom-detail.component.ts [hunk 4]
```diff
@@ -1,24 +1,52 @@
...
 
   ngOnInit(): void {
     this.activatedRoute.data.subscribe(({ symptom }) => {
       this.symptom = symptom;
     });
+    this.registerChangeInSymptom();
+  }
+
+  ngOnDestroy(): void {
+    if (this.eventSubscriber) {
```
Comment (sfdsilva) [inline src/main/webapp/app/entities/symptom/symptom-detail.component.ts:34]: There seems to be some github actions that you need to deal with. Please check this.

**Comment by andrenns** [comment]
Screenshot from 2020-12-07 21-23-33
In order to achieve 100% coverage you have to cover these line marked with a red bar, look in therapeutic regimes update tests, there are some similar line being tested there(If I'm not wrong)

**Comment by JoaoLetra** [comment]
In acceptance with @xalves, the test only covers 84% of all functions.
in symptom-detail.component.ts is missing delete() function.
symptom-delete-dialog.component.ts is missing an if condition - if (this.eventName)

**Comment by benjamim-ferreira** [comment]
after talking with @JoaoLetra he told me that @xalves (QA) had accepted the 84% of coverage.
Review completed

**Pull request merged**